source/Halibut/ServiceModel/PendingRequestQueue.cs

```diff
@@ -99,20 +99,26 @@ public int Count
 
         public async Task<RequestMessage> DequeueAsync(CancellationToken cancellationToken)
         {
-            var pending = await DequeueNextAsync();
+            var pending = await DequeueNextAsync(cancellationToken);
             if (pending == null) return null;
             return pending.BeginTransfer() ? pending.Request : null;
         }
 
-        async Task<PendingRequest> DequeueNextAsync()
+        async Task<PendingRequest> DequeueNextAsync(CancellationToken cancellationToken)
         {
             var first = TakeFirst();
             if (first != null)
             {
                 return first;
             }
 
-            await Task.WhenAny(hasItems.WaitAsync(), Task.Delay(pollingQueueWaitTimeout));
+            using var cleanupCancellationTokenSource = new CancellationTokenSource();
+            using var linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cleanupCancellationTokenSource.Token);
+            
+            await Task.WhenAny(hasItems.WaitAsync(), Task.Delay(pollingQueueWaitTimeout, linkedCancellationTokenSource.Token));
```

Comment:
For later, but we're seeing this pattern emerge. 

Would an extension method be worth making at some point that does all this for us? Again, for later.